src/main/java/com/google/sps/servlets/EditOrganizationServlet.java

```diff
@@ -18,10 +18,85 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Entity;
+import java.util.ArrayList;
+import java.util.Arrays;
 import com.google.sps.data.User;
 import java.io.IOException;
+import com.google.gson.Gson;
 
 @WebServlet("/edit-organization")
 public class EditOrganizationServlet extends HttpServlet {
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
+    //Get the proper organization entity for updating
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    long id = Long.parseLong(request.getParameter("id"));
+    Key organizationKey = KeyFactory.createKey("Distributor", id);
+    Entity organization = new Entity("Distributor");
+
+    // try catch for compliation purposes, servlet will not be called without a valid id param
+    try {
+      organization = datastore.get(organizationKey);
+    } catch(com.google.appengine.api.datastore.EntityNotFoundException err) {
+        System.out.println("Entity Not Found");
+    }
+
+    // Get all information from form
+    String newOrgName = request.getParameter("org-name");
```

Comment:
could you create a map of param -> property name?

"org-name" -> "orgName", etc

then you can write this as a loop! (at least for the properties you copy over)

for formKey, objKey:
   String value = request.getParameter(formKey);
   if (value is null) fail
   organization.setProperty(objKey, value);

Bonus points if you can figure out how to coerce "approval" into true/false in the above scheme (hint; you don't want to just store the object Key in the map)